From ffb6108d30b427cb7debe0b00d874e52362bdb47 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:41 -0400
Subject: [tools] perf/sched: Introduce --list-cmds for use by scripts

Message-id: <1410181752-25631-157-git-send-email-jolsa@redhat.com>
Patchwork-id: 91913
O-Subject: [PATCH RHEL7.1 BZ1134356 156/307] perf sched: Introduce --list-cmds for use by scripts
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit a83edb2dfc5989fbadc594109c933bae528a2809
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Fri Mar 14 23:17:54 2014 -0400
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-sched.c b/tools/perf/builtin-sched.c
index 9ac0a49..d3fb0ed 100644
--- a/tools/perf/builtin-sched.c
+++ b/tools/perf/builtin-sched.c
@@ -1713,8 +1713,10 @@ int cmd_sched(int argc, const char **argv, const char *prefix __maybe_unused)
   "perf sched replay [<options>]",
   NULL
  };
- const char * const sched_usage[] = {
-  "perf sched [<options>] {record|latency|map|replay|script}",
+ const char *const sched_subcommands[] = { "record", "latency", "map",
+        "replay", "script", NULL };
+ const char *sched_usage[] = {
+  NULL,
   NULL
  };
  struct trace_sched_handler lat_ops  = {
@@ -1736,8 +1738,8 @@ int cmd_sched(int argc, const char **argv, const char *prefix __maybe_unused)
  for (i = 0; i < ARRAY_SIZE(sched.curr_pid); i++)
   sched.curr_pid[i] = -1;
 
- argc = parse_options(argc, argv, sched_options, sched_usage,
-        PARSE_OPT_STOP_AT_NON_OPTION);
+ argc = parse_options_subcommand(argc, argv, sched_options, sched_subcommands,
+     sched_usage, PARSE_OPT_STOP_AT_NON_OPTION);
  if (!argc)
   usage_with_options(sched_usage, sched_options);
 
diff --git a/tools/perf/perf-completion.sh b/tools/perf/perf-completion.sh
index ecedab6..3356984 100644
--- a/tools/perf/perf-completion.sh
+++ b/tools/perf/perf-completion.sh
@@ -122,7 +122,7 @@ __perf_main ()
   evts=$($cmd list --raw-dump)
   __perfcomp_colon "$evts" "$cur"
  # List subcommands for perf commands
- elif [[ $prev == @(kvm|kmem|mem|lock) ]]; then
+ elif [[ $prev == @(kvm|kmem|mem|lock|sched) ]]; then
   subcmds=$($cmd $prev --list-cmds)
   __perfcomp_colon "$subcmds" "$cur"
  # List long option names
-- 
1.7.1